core/services/ocrcommon/transmitter.go

```diff
@@ -62,11 +77,11 @@ func (t *transmitter) FromAddressForTransaction() common.Address {
 	// Use Round-Robin to select the next fromAddress.
 	nextFromAddress := t.fromAddresses[t.nextFromAddressIndex]
 
-	// Only apply round-robin logic for multiple sending keys.
-	if len(t.fromAddresses) > 1 {
-		t.nextFromAddressIndex++
-		if t.nextFromAddressIndex >= len(t.fromAddresses) {
-			t.nextFromAddressIndex = 0
+	// Only apply round-robin logic for multiple sending keys and a valid keystore.
+	if len(t.fromAddresses) > 1 && t.ethKeyStore != nil {
+		fromAddress, err := (*t.ethKeyStore).GetRoundRobinAddress(t.chainID, t.fromAddresses...)
```

Comment:
This dereference is odd, I don't think it's necessary.

Reply:
The type itself is a reference, which I just found out: https://github.com/smartcontractkit/chainlink/blob/VRF-250-round-robin-sending-address-selection-in-contract-transmitter-selects-disabled-sending-key/core/services/keystore/eth.go#L60. So I'll just get rid of the pointer all together.